internal/pkg/addon/storage.go

```diff
@@ -0,0 +1,126 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package addon contains the service to manage addons.
+package addon
+
+import (
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/template"
+)
+
+const (
+	dynamoDbAddonPath = "addons/ddb/cf.yml"
+	s3AddonPath       = "addons/s3/cf.yml"
+)
+
+type storage struct {
+	ResourceName *string
+	Name         *string
+}
+
+// DynamoDB contains configuration options which fully descibe a DynamoDB table.DynamoDB
+// Implements the encoding.BinaryMarshaler interface.
+type DynamoDB struct {
+	storage
+
+	Attributes   []DDBAttribute
+	LSIs         []LocalSecondaryIndex
+	HasLSI       bool
+	SortKey      *string
+	PartitionKey *string
```

Comment:
Can we tweak thins around so that:
```go
type DynamoDB struct {
    DynamoDBProps // embed the props

    parser template.Parser
}
```
Or is there a reason why we have to re-define the fields?